File: app_abarrotes/lib/models/recepcion_compra.dart
class RecepcionCompra {
  final int id;
  final int? ordenCompraId;
  final int? proveedorId;
  final int almacenId;
  final String? numeroDocumento;
  final String? tipoDocumento;
  final String fechaRecepcion;
  final String estado;
  final String? observaciones;
  final String? ordenCompraCodigo;
  final String? proveedorNombre;
  final String? almacenNombre;

  RecepcionCompra({
    required this.id,
    this.ordenCompraId,
    this.proveedorId,
    required this.almacenId,
    this.numeroDocumento,
    this.tipoDocumento,
    required this.fechaRecepcion,
    this.estado = 'parcial',
    this.observaciones,
    this.ordenCompraCodigo,
    this.proveedorNombre,
    this.almacenNombre,
  });

  factory RecepcionCompra.fromJson(Map<String, dynamic> json) {
    return RecepcionCompra(
      id: json['id'],
      ordenCompraId: json['orden_compra_id'],
      proveedorId: json['proveedor_id'],
      almacenId: json['almacen_id'],
      numeroDocumento: json['numero_documento'],
      tipoDocumento: json['tipo_documento'],
      fechaRecepcion: json['fecha_recepcion'],
      estado: json['estado'] ?? 'parcial',
      observaciones: json['observaciones'],
      ordenCompraCodigo: json['orden_compra']?['codigo'],
      proveedorNombre: json['proveedor']?['nombre'],
      almacenNombre: json['almacen']?['nombre'],
    );
  }

  Map<String, dynamic> toJson() => {
    'orden_compra_id': ordenCompraId,
    'proveedor_id': proveedorId,
    'almacen_id': almacenId,
    'numero_documento': numeroDocumento,
    'tipo_documento': tipoDocumento,
    'fecha_recepcion': fechaRecepcion,
    'estado': estado,
    'observaciones': observaciones,
  };
}

File: backend/app/Models/RecepcionCompra.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class RecepcionCompra extends Model
{
    protected $table = 'recepciones_compra';

    /** Serie del documento formal de recepción. */
    public const SERIE = 'RC01';

    protected $fillable = [
        'orden_compra_id',
        'compra_id',
        'proveedor_id',
        'almacen_id',
        'serie',
        'numero',
        'numero_documento',
        'tipo_documento',
        'fecha_recepcion',
        'estado',
        'activo',
        'stock_aplicado',
        'usuario_recibe_id',
        'observaciones',
    ];

    protected $appends = ['documento'];

    protected function casts(): array
    {
        return [
            'fecha_recepcion' => 'datetime',
            'stock_aplicado' => 'boolean',
            'activo' => 'boolean',
        ];
    }

    /** Número formal del documento de recepción, ej. "RC01-0024". */
    public function getDocumentoAttribute(): ?string
    {
        if (! $this->serie || ! $this->numero) {
            return null;
        }

        return "{$this->serie}-{$this->numero}";
    }

    public function ordenCompra()
    {
        return $this->belongsTo(OrdenCompra::class);
    }

    public function compra()
    {
        return $this->belongsTo(Compra::class);
    }

    public function proveedor()
    {
        return $this->belongsTo(Proveedor::class);
    }

    public function almacen()
    {
        return $this->belongsTo(Almacen::class);
    }

    public function usuarioRecibe()
    {
        return $this->belongsTo(User::class, 'usuario_recibe_id');
    }

    public function detalles()
    {
        return $this->hasMany(RecepcionCompraDetalle::class, 'recepcion_id');
    }

    public function devoluciones()
    {
        return $this->hasMany(DevolucionProveedor::class);
    }
}
